packages/starknet_provider/lib/src/model/block_with_receipts.dart

```diff
@@ -20,16 +18,16 @@ class TransactionWithReceipt with _$TransactionWithReceipt {
 @freezed
 class BlockWithReceipts with _$BlockWithReceipts {
   const factory BlockWithReceipts({
-    @JsonKey(name: 'block_hash') String? blockHash,
-    @JsonKey(name: 'parent_hash') String? parentHash,
-    @JsonKey(name: 'block_number') int? blockNumber,
-    @JsonKey(name: 'sequencer_address') String? sequencerAddress,
-    @JsonKey(name: 'new_root') String? newRoot,
+    Felt? blockHash,
+    Felt? parentHash,
+    int? blockNumber,
+    Felt? sequencerAddress,
+    Felt? newRoot,
     int? timestamp,
-    @JsonKey(name: 'starknet_version') String? starknetVersion,
-    @JsonKey(name: 'l1_gas_price') GasPrice? l1GasPrice,
-    @JsonKey(name: 'l1_data_gas_price') GasPrice? l1DataGasPrice,
-    @JsonKey(name: 'l1_da_mode') String? l1DaMode,
+    String? starknetVersion,
+    GasPrice? l1GasPrice,
+    GasPrice? l1DataGasPrice,
```

Comment:
I think `GasPrice` should be renamed into `ResourcePrice` to match JSON RPC sepcification: https://github.com/starkware-libs/starknet-specs/blob/4233164155b8a279b408b9638c34db2602b5cdc0/api/starknet_api_openrpc.json#L1470-L1479